run_project_tests.py

```diff
@@ -85,7 +93,16 @@ def __exit__(self, _type, value, traceback):
         # holding files open.
         mesonlib.windows_proof_rmtree(self.dir)
 
+compile_commands = None  # type: str
+clean_commands = None  # type: str
+install_commands = None  # type: str
+uninstall_commands = None  # type: str
+backend_flags = []  # type: typing.List[str]
+executor = None  # type: ignore
+logfile = None  # type: typing.TextIO
+backend = None  # type: ignore
```

Comment:
a bunch of these annotations are wrong, they say that they're not `None`, but then are assigned `None`, they either need to be `Optional[str]` (for example), or assigned an empty value of the correct type.

Reply:
MyPy needs globals to be given some value, even None before the global is used in a function. I fixed these where practical, the ignored ones I set to None because of excessively complexity and interaction with run_test.py